### ISSUE TITLE
un-deprecate programs.neovim.configure

### ISSUE THREAD

**Issue by kalbasit** [rycee/home-manager#971]
@rycee what was the motivation for deprecating the `configure` option?

In shabka, we declare a common [neovim configuration module](https://github.com/kalbasit/shabka/blob/d02d93069fe45e5c4296f56a2ebd1096b7fb1138/modules/neovim/default.nix) that are being used directly for configuring [NixOS](https://github.com/kalbasit/shabka/blob/d02d93069fe45e5c4296f56a2ebd1096b7fb1138/modules/nixos/software/neovim.nix#L50) and [home-manager](https://github.com/kalbasit/shabka/blob/d02d93069fe45e5c4296f56a2ebd1096b7fb1138/modules/home/software/neovim.nix#L51). It would be a shame to lose this consistency.

We're currently this in https://github.com/kalbasit/shabka/issues/286

_Originally posted by @kalbasit in https://github.com/rycee/home-manager/pull/793_

**Comment by Gerschtli** [comment]
Hey, as I was the one introducing this deprecation: I think we should remove the deprecation warning and let the use decide on how he configures neovim. The reason why I added the deprecation warning was consistency between vim and neovim modules and because I thought there were no benefit in supporting the `configure` option.

As you proved me wrong and were some other concerns regarding plugin loading order of vim plugins (cannot remember the thread we discussed it), let's simply remove the deprecation warning.
